pymeasure/instruments/teledyne/teledyneT3AFG.py

```diff
@@ -0,0 +1,131 @@
+#
+# Copyright (c) 2013-2022 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+
+from pymeasure.instruments import Instrument, Channel
+from pymeasure.instruments.validators import strict_range, strict_discrete_set
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class SignalChannel(Channel):
+
+    def get_process_generator(keyword, unit, type):
+        """Generate a get_process method searching for keyword, stripping unit"""
+        def selector(values):
+            if keyword in values:
+                try:
+                    return type(values[values.index(keyword)+1].strip(unit))
+                except (ValueError, IndexError):
+                    # Something went quite wrong if the keyword exists but the value doesn't
+                    return None
+            else:
+                # Wrong wavetype for this keyword
+                return None
+        return selector
+
+    output_enabled = Channel.control(
+        "C{ch}:OUTPut?",
+        "C{ch}:OUTPut %s",
+        """Control whether the channel output is enabled (boolean).""",
+        validator=strict_discrete_set,
+        map_values=True,
+        values={True: 'ON', False: 'OFF'},
+        # Replace ON and OFF with True and False in both get and set
+        get_process=lambda x: True if x[0].split(' ')[1] == 'ON' else False,
+    )
+
+    # TODO: Add other OUTPut related controls like Load and Polarity
+
+    wavetype = Channel.control(
+        "C{ch}:BSWV?",
+        "C{ch}:BSWV WVTP,%s",
+        """Control the type of waveform to be output.
+        Options are: {SINE, SQUARE, RAMP, PULSE, NOISE, ARB, DC, PRBS, IQ}
+        """,
+        validator=strict_discrete_set,
+        values=['SINE', 'SQUARE', 'RAMP', 'PULSE', 'NOISE', 'ARB', 'DC', 'PRBS', 'IQ'],
+        get_process=lambda x: x[1],
+    )
+
+    # TODO: Add frequency ranges per model?
```

Comment:
For your information. With [dynamic properties](https://pymeasure.readthedocs.io/en/latest/dev/adding_instruments.html#dynamic-properties) you can change the ranges per module afterwards.

Reply:
I'll go ahead and preemptively make the relevant properties dynamic, but I am not planning on writing instrument definitions for other instruments in the family since I can't test them.